STM/src/main/java/org/jboss/stm/internal/optimistic/OptimisticLockRecord.java

```diff
@@ -135,7 +135,7 @@ public int topLevelCommit ()
     {       
         boolean stateOK = checkState();
         
-        if (stateOK)
+        if (!stateOK)
```

Comment:
We get the warning on the normal path where the commit will succeed. So to the casual onlooker it appears as if there is a problem when in fact all is well.